src/generateTypeFiles.js

```diff
@@ -82,12 +76,51 @@ export const generateTypeFiles = (
                 ? `// To regenerate, run '${options.regenerateCommand}'.\n`
                 : '') +
             code;
+        if (options.splitTypes) {
+            fileContents +=
+                `\nexport type ${name} = ${typeName}['response'];\n` +
+                `export type ${name}Variables = ${typeName}['variables'];\n`;
+        }
 
-        fs.writeFileSync(targetPath, fileContents);
         addToIndex(targetPath, typeName);
+        files[targetPath] = fileContents;
     });
 
+    return {files, indexContents};
+};
+
+export const generateTypeFiles = (
+    fileName: string,
+    schema: Schema,
+    document: DocumentNode,
+    options: Options,
+) => {
+    const generatedDir = path.join(
+        path.dirname(fileName),
+        options.generatedDirectory ?? '__generated__',
+    );
+    const indexFile = path.join(generatedDir, 'index.js');
+
+    if (!fs.existsSync(generatedDir)) {
+        fs.mkdirSync(generatedDir, {recursive: true});
```

Comment:
Cool. I didn't realize the built-in `mkdir` had support for recursive directories.  Time to drop `mkdirp`.